src/simple_regular_solve.jl

```diff
@@ -50,6 +50,136 @@ function DiffEqBase.solve(jump_prob::JumpProblem, alg::SimpleTauLeaping;
         interp = DiffEqBase.ConstantInterpolation(t, u))
 end
 
+struct SimpleAdaptiveTauLeaping <: DiffEqBase.DEAlgorithm
+    epsilon::Float64  # Error control parameter
+end
+
+SimpleAdaptiveTauLeaping(; epsilon=0.05) = SimpleAdaptiveTauLeaping(epsilon)
+
+function compute_gi(u, nu, hor, i)
+    max_order = 1.0
```

Comment:
Why not just make this an integer? It should just be an integer right given that `hor` is integer?

Generally try not to assume floating point types in your code. They should be inferred from the types of `t` or the type of the elements of `u` as appropriate.